internal/base_auth/auth.go

```diff
@@ -0,0 +1,73 @@
+package base_auth
```

Comment:
Random thing I'm curious about:

I wondered if it's common practice in Go world to maybe use another format for packages as opposed to snake_case. I would have _guessed_ camelCase

Buuut at least judging from two other repos we have it's (wild to me) just all lowercase?
https://github.com/datarobot/oauth-providers-service/blob/main/internal/httperror/error.go#L13
https://github.com/datarobot/nbx-git-helper/blob/main/internal/gitconfig/githelper.go#L13

Curious if there's actually a preference to do all lowercase like that 🤔 

Huh - likve tweeting my thoughts - Go says:
> Good package names are short and clear. They are lower case, with no under_scores or mixedCaps.

https://go.dev/blog/package-names